pkg/webhook/mutation/pod/metadata/workload.go

```diff
@@ -99,6 +99,9 @@ func findRootOwner(ctx context.Context, clt client.Client, partialObjectMetadata
 			}
 
 			return findRootOwner(ctx, clt, ownerObjectMetadata)
+		} else {
+			// pod is created by workload of kind that is not well known
+			return newUnknownWorkloadInfo(), nil
```

Comment:
there might be more owner references
so `return` statement here seems to be premature

Reply:
https://github.com/Dynatrace/dynatrace-operator/pull/3170/commits/d0f8b4e6a080075fff2c926bd4e2321004851a36 Is a proposed solution. I am not sure how stable it is so please if anyone has feedback please let me know.